### PR TITLE
Implement ipCidrRouteDest oid (only default routes)

### PR DESCRIPTION
**Pull request by qiluo-msft**

### CODE REVIEW

File: src/sonic_ax_impl/mibs/ietf/rfc4292.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        self.route_dest_map = {}
+        self.route_dest_list = []
+
+        self.db_conn.connect(mibs.APPL_DB)
+        ## TODO: error handling
+        ent = self.db_conn.get_all(mibs.APPL_DB, "ROUTE_TABLE:" + ipn.with_prefixlen, blocking=True)
+        nexthops = ent[b"nexthop"].decode()
+        for nh in nexthops.split(','):
+            sub_id = ip2tuple(ipn.network_address) + ip2tuple(ipn.netmask) + ip2tuple(nh)
+            # print(sub_id)
```
Comment (pavel-shirshov) [inline src/sonic_ax_impl/mibs/ietf/rfc4292.py]: Remove debug output?

File: src/sonic_ax_impl/mibs/ietf/rfc4292.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        """
+        Update redis (caches config)
+        Pulls the table references for each interface.
+        """
+        ipn = ipaddress.ip_network("0.0.0.0/0")
+        self.route_dest_map = {}
+        self.route_dest_list = []
+
+        self.db_conn.connect(mibs.APPL_DB)
+        ## TODO: error handling
```
Comment (pavel-shirshov) [inline src/sonic_ax_impl/mibs/ietf/rfc4292.py]: Should we do error handling before merge of this PR?

File: tests/test_forward.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+import ipaddress
+
+modules_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: double dirname function. Do we need this?

File: tests/test_forward.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+class TestForwardMIB(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.lut = MIBTable(SonicMIB)
+
+    # def test_print_oids(self):
+    #     for k in self.lut.keys():
+    #         print(k)
+    #     mib_entry = self.lut[(1, 3, 6, 1, 2, 1, 4, 24, 4, 1, 1)]
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: Remove it, or uncomment it?

File: tests/test_forward.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                ObjectIdentifier(20, 0, 0, 0, (1, 3, 6, 1, 2, 1, 4, 24, 4, 1, 1)),
+            )
+        )
+
+        encoded = get_pdu.encode()
+        response = get_pdu.make_response(self.lut)
+        print(response)
+
+        n = len(response.values)
+        # self.assertEqual(n, 7)
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: ?

File: tests/test_forward.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            header=PDUHeader(1, PduTypes.GET, 16, 0, 42, 0, 0, 0),
+            oids=[oid]
+        )
+
+        encoded = get_pdu.encode()
+        response = get_pdu.make_response(self.lut)
+        print(response)
+
+        n = len(response.values)
+        # self.assertEqual(n, 7)
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: ?

File: tests/test_forward.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                ObjectIdentifier(19, 0, 0, 0, (1, 3, 6, 1, 2, 1, 4, 24, 4, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1)),
+            )
+        )
+
+        encoded = get_pdu.encode()
+        response = get_pdu.make_response(self.lut)
+        print(response)
+
+        n = len(response.values)
+        # self.assertEqual(n, 7)
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: ?

File: tests/test_forward.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                ObjectIdentifier(11, 0, 0, 0, (1, 3, 6, 1, 2, 1, 4, 24, 4, 1, 2)),
+            )
+        )
+
+        encoded = get_pdu.encode()
+        response = get_pdu.make_response(self.lut)
+        print(response)
+
+        n = len(response.values)
+        # self.assertEqual(n, 7)
```
Comment (pavel-shirshov) [inline tests/test_forward.py]: ?

File: tests/test_forward.py (hunk 3 above)
Comment (qiluo-msft) [inline tests/test_forward.py]: The outmost one is to get the parent dirname. It is needed.

---
In reply to: [99025224](https://github.com/Azure/sonic-snmpagent/pull/4#discussion_r99025224) [](ancestors = 99025224)

**Comment by qiluo-msft** [comment]
@pavel-shirshov Resolved all. Could you review again?

**Pull request merged**